File: python/make_data_key.py
#
# Copyright 2019-present MongoDB, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

"""This file finds and prints or makes and prints a data key used for
encrpytion."""

from helpers import read_master_key, CsfleHelper


def main():

    local_master_key = read_master_key()

    kms_providers = {
        "local": {
            "key": local_master_key,
        },
    }

    csfle_helper = CsfleHelper(kms_providers=kms_providers)
    data_key = csfle_helper.find_or_create_data_key()

    print("Base64 data key. Copy and paste this into app.py\t", data_key)


if __name__ == "__main__":
    main()

File: nodejs/make-data-key.js
const kms = require("./kms");
require("dotenv").config();

async function main(unencryptedClient) {
  try {
    const kmsClient = kms.localCsfleHelper();

    unencryptedClient = await kmsClient.getRegularClient();

    const dataKey = await kmsClient.findOrCreateDataKey(unencryptedClient);
    console.log(
      `Base64 data key. Copy and paste this into clients.js:

${dataKey}
`
    );
  } finally {
    if (unencryptedClient) await unencryptedClient.close();
  }
}
let unencryptedClient = null;
main(unencryptedClient).catch(console.dir);
